syntax = "proto3";

package iroha.protocol;

import "block.proto";
import "queries.proto";
import "google/protobuf/empty.proto";
import "responses.proto";


enum StatelessValidation {
  STATELESS_VALIDATION_FAILED = 0;
  STATELESS_VALIDATION_SUCCESS = 1;
}

message ToriiResponse {
  StatelessValidation validation = 1;
}

service CommandService {
  rpc Torii (Transaction) returns (ToriiResponse);
}


service QueryService {
  rpc Find (Query) returns (QueryResponse);
}

enum GenesisBlockApplied {
  APPLY_FAILURE = 0;
  APPLY_SUCCESS = 1;
}

message ApplyGenesisBlockResponse {
  GenesisBlockApplied applied = 1;
}

service GenesisBlockService {
  rpc SendGenesisBlock (Block) returns (ApplyGenesisBlockResponse);
  rpc SendAbortGenesisBlock (Block) returns (google.protobuf.Empty);
}
